cytotable/convert.py

```diff
@@ -152,17 +153,36 @@ def _source_chunk_to_parquet(
 
     result_filepath = f"{result_filepath_base}-{offset}.parquet"
 
-    # isolate using new connection to read data with chunk size + offset
-    # and export directly to parquet via duckdb (avoiding need to return data to python)
-    _duckdb_reader().execute(
-        f"""
-        COPY (
-            {base_query}
-            LIMIT {chunk_size} OFFSET {offset}
-        ) TO '{result_filepath}'
-        (FORMAT PARQUET);
-        """
-    )
+    # attempt to read the data to parquet from duckdb
+    # with exception handling to read mixed-type data
+    # using sqlite3 and special utility function
+    try:
+        # isolate using new connection to read data with chunk size + offset
+        # and export directly to parquet via duckdb (avoiding need to return data to python)
+        _duckdb_reader().execute(
+            f"""
+            COPY (
+                {base_query}
+                LIMIT {chunk_size} OFFSET {offset}
+            ) TO '{result_filepath}'
+            (FORMAT PARQUET);
+            """
+        )
+    except duckdb.Error as e:
```

Comment:
Thanks for calling this forward! I noticed mixed behavior with DuckDB Python Exceptions which resulted in the approach here. I'll give the more specific exception another attempt here (very much prefer it! 🙂 ).

Reply:
Digging into this more I found that only generic errors are raised from the context of when this happens from the `sqlite_scanner` extension. I'm going to raise an issue for that extension with the hope that we may get more feedback here. In the meantime, I'll plan to leave things as-is.

See the following demonstration code:

```python
import sqlite3

import duckdb

# create a temporary sqlite connection
filepath = ".example.sqlite"

# statements for creating database with simple structure
create_stmts = """
    DROP TABLE IF EXISTS tbl_a;
    CREATE TABLE tbl_a (
        col_integer INTEGER NOT NULL
        ,col_text TEXT
        ,col_blob BLOB
        ,col_real REAL
    );
    """

# some example values to insert into the database
insert_vals = ["nan", "sample", b"sample_blob", 0.5]

# create the database and insert some data into it
with sqlite3.connect(filepath) as connection:
    connection.executescript(create_stmts)

    connection.execute(
        (
            "INSERT INTO tbl_a (col_integer, col_text, col_blob, col_real)"
            "VALUES (?, ?, ?, ?);"
        ),
        insert_vals,
    )

try:
    duckdb.connect().execute(
        f"""
    /* install and load sqlite plugin for duckdb */
    INSTALL sqlite_scanner;
    LOAD sqlite_scanner;

    /* perform query on tbl_a table */
    SELECT * FROM sqlite_scan('{filepath}', 'tbl_a');
    """
    ).arrow()
    
except duckdb.TypeMismatchException as type_mismatch_e:
    print("Found a type mismatch exception: ", type_mismatch_e)
    
except Exception as e:
    print("Found a different error of type: ", type(e), e, sep="\n")
```

Reply:
Nice detective work, and a pity the extension isn't throwing the correct exception. Sounds good regarding leaving things as-is for now, since there aren't any better options.